net/icmpv6/icmpv6_input.c

```diff
@@ -353,7 +353,7 @@ void icmpv6_input(FAR struct net_driver_s *dev, unsigned int iplen)
                   {
                     FAR struct icmpv6_mtu_s *mtuopt =
                                         (FAR struct icmpv6_mtu_s *)opt;
-                    dev->d_pktsize = NTOHL(mtuopt->mtu) + dev->d_llhdrlen;
+                    dev->d_pktsize = NTOHS(mtuopt->mtu[1]) + dev->d_llhdrlen;
```

Comment:
what about `mtu[0]`? Is it always 0?

Reply:
All our supported device types will be 0 on higher bits. Meanwhile, our d_pktsize is uint16_t, so there's no difference even if it's not 0 (and we cannot handle) ☺️

Reply:
Yes. I checked the `d_pktsize` type after I posted a comment :)